perfkitbenchmarker/benchmarks/cloudsuite_web_serving.py

```diff
@@ -0,0 +1,411 @@
+                         ' %s/etc/config.php'
+                         % ('(.*olioconfig.*cacheSystem.*MemCached.*)',
+                            '\/\/\$1', APP_DIR))
+  frontend.RemoteCommand('perl -pi -e '
+                         '"s/%s/%s/g"'
+                         ' %s/etc/config.php'
+                         % ('\/\/(.*olioconfig.*cacheSystem.*NoCache.*)',
+                            '\$1', APP_DIR))
+  frontend.RemoteCommand('perl -pi -e '
+                         '"s/%s/%s/g"'
+                         ' %s/etc/config.php'
+                         % ('GEOCODER_HOST', backend.ip_address, APP_DIR))
+  frontend.Install('php')
+  php.ConfigureAndBuild(frontend, PHPRC, True)
+  frontend.RemoteCommand('mkdir -p /tmp/http_sessions && '
+                         'chmod 777 /tmp/http_sessions')
+  frontend.RemoteCommand('echo """extension_dir=\\"/usr/local/lib/php/'
+                         'extensions/no-debug-non-zts-20090626/\\""""'
+                         ' >> %s/php.ini'
+                         % PHPRC)
+  frontend.RemoteCommand('echo """date.timezone=\\"Europe/Zurich\\"""" >> '
+                         '%s/php.ini'
+                         % PHPRC)
+  frontend.RemoteCommand('perl -pi -e '
+                         '"s/%s/%s/g"'
+                         ' %s/php.ini'
+                         % ('(display_errors).*', '\$1 = Off', PHPRC))
+  frontend.RemoteCommand('perl -pi -e '
+                         '"s/%s/%s/g"'
+                         ' %s/php.ini'
+                         % ('(error_reporting).*', '\$1 = E_ALL & ~E_NOTICE',
+                            PHPRC))
+  php.InstallAPC(frontend)
+
+  def setupFilestore(vm):
+    filestore = posixpath.join(vm.GetScratchDir(), 'filestore')
+    vm.RemoteCommand('cd %s && '
+                     'patch -p1 -t < %s/cloudsuite.patch && '
+                     'mkdir -p %s && '
+                     'chmod a+rwx %s && '
+                     'chmod +x %s/benchmarks/OlioDriver/bin/fileloader.sh'
+                     % (APP_DIR, BASE_DIR, filestore, filestore,
+                        FABAN_HOME))
+    vm.RemoteCommand('export FILESTORE=%s && '
+                     'export JAVA_HOME=%s && '
+                     '%s/benchmarks/OlioDriver/bin/fileloader.sh 102 %s'
+                     % (filestore, JAVA_HOME, FABAN_HOME, filestore))
+    frontend.RemoteCommand('perl -pi -e '
+                           '"s/%s/%s/g"'
+                           ' %s/etc/config.php'
+                           % ('(\$olioconfig\[\'localfsRoot\'\]).*',
+                              '\$1 = \'%s\';' % re.escape(filestore),
+                              APP_DIR))
+    return
+
+  setupFilestore(frontend)
+  frontend.RemoteCommand('sudo cp /usr/local/etc/php-fpm.conf.default '
+                         '/usr/local/etc/php-fpm.conf && '
+                         'sudo /usr/local/sbin/php-fpm')
+  nginx.Start(frontend, benchmark_spec.firewall)
+  return
+
+
+def setupBackend(benchmark_spec):
```

Comment:
These three setup functions are pretty complicated. Would it make more sense to separate them out into their own scripts in the data directory and have the benchmark copy the scripts to the VM and execute them?